test/blackbox/run_blackbox_tests/test_cases_key_rotation.c

```diff
@@ -0,0 +1,514 @@
+
+    Expected Result:
+    Terminating node instance when meshlink_encrypted_key_rotate function called
+    at any stage should give atomic result when reopened.
+*/
+static bool test_key_rotation_05(void) {
+	pid_t pid;
+	int status;
+	meshlink_handle_t *mesh;
+	meshlink_destroy("encrypted_conf");
+	meshlink_set_log_cb(NULL, MESHLINK_DEBUG, log_cb);
+
+	assert(signal(SIGINT, SIG_DFL) != SIG_ERR);
+	assert(signal(SIGABRT, SIG_DFL) != SIG_ERR);
+
+	// Set debug_probe callback
+
+	devtool_keyrotate_probe = debug_probe;
+	int new_port = 12000;
+	int pipefd[2];
+
+	// incrementally debug meshlink_encrypted_key_rotate API atomicity
+
+	for(break_stage = 1; break_stage <= 3; break_stage += 1) {
+		fprintf(stderr, "Debugging stage %d\n", break_stage);
+		meshlink_destroy("encrypted_conf");
+
+		assert(pipe(pipefd) != -1);
+
+		pid = fork();
+		assert(pid != -1);
+
+		if(!pid) {
+			close(pipefd[0]);
+			mesh = meshlink_open_encrypted("encrypted_conf", "foo", "encrypted", DEV_CLASS_BACKBONE, "oldkey", 6);
+			assert(mesh);
+			meshlink_set_log_cb(mesh, MESHLINK_DEBUG, log_cb);
+			meshlink_enable_discovery(mesh, false);
+
+			assert(meshlink_set_port(mesh, new_port));
+
+			char *invitation = meshlink_invite(mesh, NULL, "bar");
+			assert(invitation);
+
+			assert(write(pipefd[1], invitation, strlen(invitation) + 1) != -1);
+
+			meshlink_encrypted_key_rotate(mesh, "newkey", 6);
+			raise(SIGABRT);
+		}
+
+		close(pipefd[1]);
+
+		// Wait for child exit and verify which signal terminated it
+
+		assert(waitpid(pid, &status, 0) != -1);
+		assert_int_equal(WIFSIGNALED(status), true);
+		assert_int_equal(WTERMSIG(status), SIGINT);
+
+		// Reopen the node with invalid key other than old and new key should fail and should not affect
+		// the existing confbase
+
+		fprintf(stderr, "Opening mesh with invalid key\n");
+		mesh = meshlink_open_encrypted("encrypted_conf", "foo", "encrypted", DEV_CLASS_BACKBONE, "invalidkey", 9);
+		assert_int_equal(mesh, NULL);
+
+		// Reopen the node with the "newkey", if it failed to open with "newkey" then
+		// opening with the "oldkey" should succeed
+
+		fprintf(stderr, "Opening mesh with new-key\n");
+		mesh = meshlink_open_encrypted("encrypted_conf", "foo", "encrypted", DEV_CLASS_BACKBONE, "newkey", 6);
+
+		if(!mesh) {
+			fprintf(stderr, "Opening mesh with new-key failed trying to open with old-key\n");
+			mesh = meshlink_open_encrypted("encrypted_conf", "foo", "encrypted", DEV_CLASS_BACKBONE, "oldkey", 6);
+			assert_int_not_equal(mesh, NULL);
+		}
+
+		meshlink_set_log_cb(mesh, MESHLINK_DEBUG, log_cb);
+		meshlink_enable_discovery(mesh, false);
+
+		// Verify the newly set port and generated invitation
+
+		int get_port = meshlink_get_port(mesh);
+		assert_int_equal(get_port, new_port);
+
+		char invitation[200];
+		assert(read(pipefd[0], invitation, sizeof(invitation)) != -1);
+
+		assert(meshlink_start(mesh));
+
+		meshlink_destroy("encrypted_conf.1");
+
+		set_sync_flag(&status_changed_cond, false);
```

Comment:
This fails to compile, `status_changed_cond` is not defined anywhere.

Reply:
I need to delete those parameters which I resolved now & force pushed.